src/CachedResponseClient.php

```diff
@@ -60,7 +60,11 @@ public function sendRequest(RequestInterface $request)
             $url = str_replace($this->apiKey, '[apikey]', $url);
         }
 
-        $file = sprintf('%s/%s_%s', $this->cacheDir, $host, sha1($url));
+        $cacheKey = $url;
+        if ('POST' == $request->getMethod()) {
+            $cacheKey .= $request->getBody();
```

Comment:
What if the body is 1Mb large...? Could we do this in a smarter way?

Reply:
You have to base the hash on the whole body if you want to cache the POST request. Even if the body is 1MB large. I do not see any way around that. Or do you have anything in mind?

Reply:
Also, I do not think we will ever have 1MB large bodys for any request.

Reply:
I did not think we would have POST requests at all.. =)

A solution that may work is to include the body length only. But that is not 100%...